#161397756 Implementation of LFA's feedback

abejide001 opened the pull request:
**What does this PR do**
Implement LFA's feedback on the back-end based on the code review.
**Description of tasks completed**
1. Replaced express.json() with bodyparser.json()
2. Moved routes folder into the src folder
3. Add message property in the body of the response when sending a POST requests to sales and products endpoints
![screenshot from 2018-10-23 18-42-22](https://user-images.githubusercontent.com/6943256/47379697-8addf880-d6f3-11e8-9a33-13bfcb06b8b7.png)

4. Re-wrote my validation, and removed the third party npm module i used for validation
5. Refractored test script in package.json
6. Rename method names, in sales and products controllers
7. Added a 200 status code on GET /products and /sales endpoint
8. Removed no-console rule from eslint
9. Re-wrote test for the sale controller, sale model, product controller

abejide001 committed:
[chore #161397756] Move routes folder into the src folder

abejide001 committed:
[chore #161397756] Remove no-console rule in eslint config

abejide001 committed:
[chore #161397756] Add before_script to .travis.yml config

abejide001 committed:
[chore #161397756] Use bodyparser to pass in data into the response body

abejide001 committed:
[chore #161397756] Create helper folder to handle validation logic for sale and product

abejide001 committed:
[chore #161397756] Refractor test script

abejide001 committed:
[chore #161397756] Modify create method and error checking  in product controller

abejide001 committed:
[chore #161397756] Remove joi that was use for validation, and change validation process

abejide001 committed:
[chore #161397756] Rename method names to be more clear

abejide001 committed:
[chore #161397756] Rewrite test for endpoints in product controller  and for the product model

abejide001 committed:
[chore #161397756] Rewrite test for endpoints in sale controller and for the sale model

abejide001 committed:
[chore #161397756] Remove validation with third party npm module, and rewrite validation

abejide001 committed:
[chore #161397756] Remove unncessary folders

abejide001 committed:
[chore #161397756] Add build for heroku

abejide001 committed:
[chore #161397756] Add build for heroku

abejide001 committed:
[chore #161397756] Modify gitignore by adding build folder

emp-daisy commented on server/src/helper/product-helper.js:
```diff
@@ -0,0 +1,45 @@
+      return validatedProduct;
+    }
+    if (validatedProduct.value.price < 1000) {
+      validatedProduct.errors.push('"price" must be larger than or equal to 1000');
```
Any reason why the price should be more than 1000?

abejide001 replied:
The price should be more than 1000, because on the front-end, no product is less than 1000
![screenshot from 2018-10-24 10-52-25](https://user-images.githubusercontent.com/6943256/47422636-01720900-d77b-11e8-9442-9bedc399426a.png)

abejide001 committed:
Implement LFA's corrections

abejide001 committed:
[chore #161397756] Refractor scripts

abejide001 committed:
[chore #161397756] Rewrite test for product model

abejide001 committed:
[chore #161397756] Remove --ignore _build from eslint script rule

abejide001 committed:
[chore #161397756] Remove _build from gitignore

abejide001 committed:
[chore #16139775] Add script to package.json

abejide001 committed:
[chore #16139775] Add npm run coverage and npm run report-change

abejide001 committed:
[chore #16139775] Add npm run coveralls to travis

abejide001 committed:
[chore #16139775] Add coveralls to script

abejide001 committed:
[chore #161397756] Add npm run coverage and npm run report coverage to travis

abejide001 committed:
[chore #161397756] Add script in packages.json

[chore #161397756] Add npm run coveralls to travis

[chore #161397756] Add script in packages.json

[chore #161397756] Write more test for product and product model

[chore #161397756] Write more test for product and product model
